include/autoppl/expression/distribution/discrete.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <cassert>
+#include <random>
+#include <cmath>
+#include <numeric>
+
+namespace ppl {
+
+// TODO: change name to DiscreteDist and make class template.
+// Discrete should be a function that creates this kind of object.
+
+template <typename weight_type>
+struct Discrete 
+{
+    using value_t = int;
+    using dist_value_t = double;
+
```

Comment:
A default constructor may be good also. The standard implements the default ctor to always produce 0. Since we sample using standard library, it may be good to follow their convention.